modules/scheduler/scheduler/src/main/resources/webapp/js/controllers.js

```diff
@@ -118,5 +129,172 @@
+            { displayName: "Sunday", value: 6}
+        ];
+
+        function containsKey(map, key) {
+            var result = false;
+            angular.forEach($scope.motechEventParameters, function(element) {
+                if (element.key === key) {
+                    result = true;
+                }
+            });
+            return result;
+        };
+
+        $scope.addToMap = function(key, value) {
+            if (containsKey($scope.motechEventParameters, key)) {
+                motechAlert("scheduler.keyAlreadyExists", "schedulerKeyAlreadyExists");
+            } else {
+                $scope.motechEventParameters.push({
+                    "key": key,
+                    "value": value
+                });
+            }
+            $timeout(function() {
+                $scope.key = "";
+                $scope.value = "";
+            });
+
+        };
+
+        $scope.resetMap = function() {
+            motechConfirm("scheduler.confirm.resetMap", "scheduler.confirm", function(response) {
+                if (response) {
+                    $timeout(function() {
+                        $scope.motechEventParameters = [];
+                    });
+                }
+            });
+        }
+
+        $scope.removeFromMap = function(key) {
+            motechConfirm("scheduler.confirm.removeItem", "scheduler.confirm", function(response) {
+                if (response) {
+                    var id;
+                    for (var i = 0; i < $scope.motechEventParameters.length; i += 1) {
+                        if ($scope.motechEventParameters[i].key === key) {
+                            id = i;
+                        }
+                    }
+                    if (id > -1) {
+                        $timeout(function () {
+                            $scope.motechEventParameters.splice(id, 1);
+                        });
+                    }
+                }
+            });
+        };
+
+        $scope.parseToDateTime = function(date) {
+            if (date && date !== "") {
+                var parts = date.split(" ");
+                return parts[0] + "T" + parts[1] + parts[2];
+            }
+            return date;
+        }
+
+        $scope.createOrUpdateJob = function(action) {
+            var job = {};
+
+            job.motechEvent = {};
+
+            for (var field in $scope.job) {
+                job[field] = $scope.job[field];
+            }
+
+            job.motechEvent.parameters = {};
+
+            angular.forEach($scope.motechEventParameters, function(parameter) {
+                job.motechEvent.parameters[parameter.key] = parameter.value;
+            });
+
+            if (job.startDate) {
+                job.startDate = $scope.parseToDateTime(job.startDate);
+            }
+
+            if (job.endDate) {
+                job.endDate = $scope.parseToDateTime(job.endDate);
+            }
+
+            if (job.days) {
```

Comment:
why check if this exists if we immediately overwrite it on line 240? shouldn't we be checking $scope.job.days here?